crates/sui-types/src/crypto.rs

```diff
@@ -927,19 +927,11 @@ pub trait SuiSignatureInner: Sized + ToFromBytes + PartialEq + Eq + Hash {
     const LENGTH: usize = Self::Sig::LENGTH + Self::PubKey::LENGTH + 1;
     const SCHEME: SignatureScheme = Self::PubKey::SIGNATURE_SCHEME;
 
-    fn get_verification_inputs(&self, author: SuiAddress) -> SuiResult<(Self::Sig, Self::PubKey)> {
-        // Is this signature emitted by the expected author?
-        let bytes = self.public_key_bytes();
-        let pk = Self::PubKey::from_bytes(bytes)
+    /// Returns the deserialized signature and deserialized pubkey.
+    fn get_verification_inputs(&self) -> SuiResult<(Self::Sig, Self::PubKey)> {
+        let pk = Self::PubKey::from_bytes(self.public_key_bytes())
             .map_err(|_| SuiError::KeyConversionError("Invalid public key".to_string()))?;
 
-        let received_addr = SuiAddress::from(&pk);
-        if received_addr != author {
-            return Err(SuiError::IncorrectSigner {
-                error: format!("Signature get_verification_inputs() failure. Author is {author}, received address is {received_addr}")
-            });
-        }
```

Comment:
this check is moved to verify_generic_secure instead bc sui address is no longer assumed to be derived from pubkey. for zklogin it can be derived from address seed and iss and sub claim name